compiler/cpp/src/thrift/generate/t_go_generator.cc

```diff
@@ -1851,6 +1864,65 @@ void t_go_generator::generate_go_struct_writer(ostream& out,
   }
 }
 
+void t_go_generator::generate_go_struct_equals(ostream& out,
+                                               t_struct* tstruct,
+                                               const string& tstruct_name,
+                                               bool is_args_or_result) {
+  if (is_args_or_result) {
+    return;
+  }
```

Comment:
nit: I think we should do this check on the call site instead, and currently there's only one caller of it (Line 1497), so there should be:

```
if !is_result && !is_args {
  generate_go_struct_equals(out, tstruct, tstruct_name);
}
```

Semantically this function is called "generate_*", but when one of its arg is passed in true it doesn't generate anything, which feels wrong to me.

Reply:
looks reasonable, thanks for suggestion.